features/enlinkd/org.opennms.features.enlinkd.shell/src/main/java/org/opennms/features/enlinkd/shell/GenerateTopologyCommand.java

```diff
@@ -0,0 +1,122 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2019 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2019 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.features.enlinkd.shell;
+
+import java.io.IOException;
+import java.sql.SQLException;
+
+import org.apache.karaf.shell.api.action.Action;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.Option;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.opennms.enlinkd.generator.TopologyGenerator;
+import org.opennms.enlinkd.generator.TopologyPersister;
+import org.opennms.enlinkd.generator.TopologySettings;
+import org.opennms.netmgt.dao.api.GenericPersistenceAccessor;
+
+/**
+ * Generate a enlinkd topology via karaf command.
+ * Install: feature:install opennms-enlinkd-shell
+ */
+@Command(scope = "enlinkd", name = "generate-topology", description = "Creates a linkd topology")
+@Service
+public class GenerateTopologyCommand implements Action {
+
+    @Option(name = "--nodes", description = "generate <N> OmnsNodes.")
+    private Integer amountNodes;
+
+    @Option(name = "--elements", description = "generate <N> CdpElements")
+    private Integer amountElements;
+
+    @Option(name = "--links", description = "generate <N> CdpLinks")
```

Comment:
Are these always CDP links? They don't seem to be. If so, I'd call it --cdp-links rather than --links.

Reply:
no, they depend on the protocol. I changed the description.